public/app/plugins/panel/alertlist/GroupByWithLoading.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, useEffect, useMemo } from 'react';
+import { isEmpty, uniq } from 'lodash';
+import { Icon, MultiSelect } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { useDispatch } from 'react-redux';
+import { fetchAllPromRulesAction } from 'app/features/alerting/unified/state/actions';
+import { useUnifiedAlertingSelector } from 'app/features/alerting/unified/hooks/useUnifiedAlertingSelector';
+import { getAllRulesSourceNames } from 'app/features/alerting/unified/utils/datasource';
+import { PromRuleType } from 'app/types/unified-alerting-dto';
+import { AlertingRule } from 'app/types/unified-alerting';
+import { isPrivateLabel } from './util';
+import {
+  isAsyncRequestMapSliceFulfilled,
+  isAsyncRequestMapSlicePending,
+} from 'app/features/alerting/unified/utils/redux';
+
+interface Props {
+  id: string;
+  defaultValue: SelectableValue<string>;
+  onChange: (keys: string[]) => void;
+}
+
+export const GroupBy: FC<Props> = (props: Props) => {
```

Comment:
nit: I don't believe you need to type props here?
```suggestion
export const GroupBy: FC<Props> = (props) => {
```